Leanplum-SDK/Classes/MessageTemplates/LPAlertMessageTemplate.m

```diff
@@ -2,46 +2,89 @@
-    [super defineActionWithContexts:contexts];
+@synthesize context;
+
++ (void)defineAction
+{
+    // Alert
     [Leanplum defineAction:LPMT_ALERT_NAME
                     ofKind:kLeanplumActionKindMessage | kLeanplumActionKindAction
              withArguments:@[
-                             [LPActionArg argNamed:LPMT_ARG_TITLE withString:APP_NAME],
-                             [LPActionArg argNamed:LPMT_ARG_MESSAGE withString:LPMT_DEFAULT_ALERT_MESSAGE],
-                             [LPActionArg argNamed:LPMT_ARG_DISMISS_TEXT withString:LPMT_DEFAULT_OK_BUTTON_TEXT],
-                             [LPActionArg argNamed:LPMT_ARG_DISMISS_ACTION withAction:nil]
-                             ]
+                 [LPActionArg argNamed:LPMT_ARG_TITLE withString:APP_NAME],
+                 [LPActionArg argNamed:LPMT_ARG_MESSAGE withString:LPMT_DEFAULT_ALERT_MESSAGE],
+                 [LPActionArg argNamed:LPMT_ARG_DISMISS_TEXT withString:LPMT_DEFAULT_OK_BUTTON_TEXT],
+                 [LPActionArg argNamed:LPMT_ARG_DISMISS_ACTION withAction:nil]
+             ]
              withResponder:^BOOL(LPActionContext *context) {
-                 @try {
-                     [[self visibleViewController] presentViewController:[self viewControllerWithContext:context] animated:YES completion:nil];
-                     [self.contexts addObject:context];
-                     return YES;
-                 }
-                 @catch (NSException *exception) {
-                     LOG_LP_MESSAGE_EXCEPTION;
-                     return NO;
-                 }
-             }];
+        @try {
+            UIAlertController *alertViewController = [UIAlertController alertControllerWithTitle:NSLocalizedString([context stringNamed:LPMT_ARG_TITLE], nil)
+                                                                                         message:NSLocalizedString([context stringNamed:LPMT_ARG_MESSAGE], nil)
+                                                                                  preferredStyle:UIAlertControllerStyleAlert];
 
-}
+            UIAlertAction *dismiss = [UIAlertAction actionWithTitle:NSLocalizedString([context stringNamed:LPMT_ARG_DISMISS_TEXT], nil)
+                                                              style:UIAlertActionStyleDefault
+                                                            handler:^(UIAlertAction *action) {
+                [context runActionNamed:LPMT_ARG_DISMISS_ACTION];
+            }];
+            [alertViewController addAction:dismiss];
+
+            UIViewController *rootViewController = [UIApplication sharedApplication].keyWindow.rootViewController;
+            [rootViewController presentViewController:alertViewController animated:YES completion:nil];
+            return YES;
+        }
+        @catch (NSException *exception) {
+            LOG_LP_MESSAGE_EXCEPTION;
+            return NO;
+        }
+    }];
+
+    // Confirm
+    [Leanplum defineAction:LPMT_CONFIRM_NAME
```

Comment:
Why are we going back to putting multiple messages in the same file? This makes it hard to get to the right file without previously having context, which is why we started breaking them up into individual files

Reply:
They are defined in same place as both are using same logic and same ui, just presenting UIAlertViewController, having 4 files to do that doesn't really make sense.